repos/system_upgrade/el7toel8/actors/kernel/checkkernelmodules/actor.py

```diff
@@ -0,0 +1,52 @@
+from leapp.actors import Actor
+from leapp.libraries.common.reporting import report_with_remediation
+from leapp.models import ActiveKernelModulesFacts, WhitelistedKernelModules
+from leapp.reporting import Report
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+
+
+class CheckKernelModules(Actor):
+    """
+    Actor checks if any loaded RHEL 7 kernel module is missing in the RHEL 8.
+    If yes, the upgrade process will be inhibited.
+
+    Inhibition is done because missing kernel module on the RHEL 8 system may
+    mean that the hardware using such module as a driver would not work on the
+    RHEL 8.
+
+    Note:
+     - List of kernel modules missing on the RHEL 8 system is located in the
+        files/removed_modules.txt file.
+    """
+    name = "check_kernel_modules"
+    consumes = (ActiveKernelModulesFacts, WhitelistedKernelModules)
+    produces = (Report,)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        with open("files/removed_modules.txt", "r") as removed:
+            removed_modules = [x.strip() for x in removed.readlines()]
+            whitelisted_modules = set()
+            modules_to_report = set()
+
+            for fact in self.consume(WhitelistedKernelModules):
+                for module in fact.whitelisted_modules:
+                    whitelisted_modules.add(module)
+
+            for fact in self.consume(ActiveKernelModulesFacts):
+                for active_module in fact.kernel_modules:
+                    if active_module.filename in removed_modules and \
```

Comment:
Instead of a complex check like this I could imagine this:
```python
            for fact in self.consume(ActiveKernelModulesFacts):
                for active_module in fact.kernel_modules:
                    if activate_module.filename in whitelisted_modules:
                        continue
                    if active_module.filename in removed_modules:
                        modules_to_report.add(active_module.filename)
```

But this is up to you

Reply:
I like this. It is cleaner.

Reply:
(Just thinking aloud)
If we had create_lookup moved out of libraries/rpms.py to a more general place (leapp libraries? some general library in leapp-repository) you could theoretically form a lookup and use set difference
```
# currently it's here so importing from rpms
from leapp.libraries.common.rpms import create_lookup
...
active_modules = create_lookup(ActiveKernelModulesFacts, 'kernel_modules', 'filename')
modules_to_report = active_modules - (set(removed_modules) - set(whitelisted_modules))
```